### PR TITLE
Add pod_template_file as that is the only way to use K8sExecutor in 2.0

### PR DESCRIPTION
**Pull request by dimberman**
## Description

> Describe the purpose of this pull request.

## PR Title

> Please make the PR title informative to the user experience. For example when adding a feature, say "Add metrics tab for Deployments" instead of "Bump Astro-UI to version X.Y.Z". Bugs are sometimes harder to make customer-facing because they are very detailed, but do what you can. Instead of "Fix metrics cardinality bug" (user does not know what 'cardinality' means and does not matter to them), say "Fix bug slowing down the metrics tab". The PR title will be used for the commit message when it's merged, and these commit messages are used to generate customer-facing release notes.

## 🎟 Issue(s)

Resolves astronomer/issues#XXXX

## 🧪  Testing

> What are the main risks associated with this change, and how are they addressed by tests added in this PR?

> Please add helm unit testing, if applicable. The docs are regretfully sparse for helm unit testing, [here](https://github.com/astronomer/helm-unittest/blob/main/DOCUMENT.md) is what we have to work with. In general, these kind of tests can be used for specific assertions like 'when these values are set, the resource ends up looking like XYZ' but not for generalized assertions like 'for all resources, make sure the namespace is not set to the release name'.

> Please also add to the system testing [here](../tests/functional-tests). This kind of testing allows you to connect into any live, running pod to make assertions about how they are operating.

## 📸 Screenshots

> Add screenshots to illustrate the changes, if applicable.

## 📋 Checklist

- [ ] The PR title is informative to the user experience
- [ ] Functional test(s) added
- [ ] Helm chart unit test(s)

### CODE REVIEW

**Comment by dimberman** [comment]
@ernest-kr this should allow you to run k8sexecutor 2.0

**Review by danielhoherd — APPROVED** [review]

File: templates/configmap.yaml [hunk 1]
```diff
@@ -73,8 +73,6 @@ data:
     worker_container_repository = {{ .Values.images.airflow.repository }}
     worker_container_tag = {{ .Values.images.airflow.tag }}
     worker_container_image_pull_policy = {{ .Values.images.airflow.pullPolicy }}
-    worker_service_account_name = {{ .Release.Name }}-worker-serviceaccount
-    image_pull_secrets = {{ template "registry_secret" . }}
```
Comment (schnie) [inline templates/configmap.yaml:77]: @dimberman why'd we make these changes?

**Comment by schnie** [comment]
@dimberman might need a rebase with main. Some of those changes should already be in there for 2.0.

File: templates/configmap.yaml (hunk 1 above)
Comment (dimberman) [reply templates/configmap.yaml:77]: @schnie these configs no longer matter when there is a pod_template_file, that said I could keep them in if we plan to keep using this chart for earlier airflow versions? (or maybe we might need to fork?)

Comment (schnie) [reply templates/configmap.yaml:77]: We could either keep them or use the `semVerCompare` function here too. What do you think?

Comment (dimberman) [reply templates/configmap.yaml:77]: @schnie that works I'll set that up now.

File: templates/configmap.yaml [hunk 2]
```diff
@@ -73,25 +73,28 @@ data:
...
-    AIRFLOW__CORE__FERNET_KEY = {{ printf "%s=fernet-key" (include "fernet_key_secret" .) }}
-{{- if .Values.elasticsearch.enabled }}
-    AIRFLOW__ELASTICSEARCH__ELASTICSEARCH_HOST = {{ printf "%s=connection" (include "elasticsearch_secret" .) }}
-{{- end }}
-
+    {{- if semverCompare ">=1.10.12" .Values.airflowVersion }}
+    pod_template_file = {{ include "airflow_pod_template_file" . }}/pod_template_file.yaml
+    {{- else }}
+    worker_service_account_name = {{ .Release.Name }}-worker-serviceaccount
+    image_pull_secrets = {{ template "registry_secret" . }}
```
Comment (schnie) [inline templates/configmap.yaml:82]:
```suggestion
    image_pull_secrets = {{ template "registry_secret" . }}

```

**Pull request merged**